model_analyzer/perf_analyzer/perf_analyzer.py

```diff
@@ -291,163 +306,35 @@ def _parse_output(self, metrics):
-
-    def _parse_perf_throughput(self, section):
-        """
-        Parses throughput from the perf analyzer output
-        """
-
-        throughput = re.search('Throughput: (\d+\.\d+e\+\d+|\d+\.\d+|\d+)',
-                               section)
-        if throughput:
-            throughput = float(throughput.group(1))
-            return PerfThroughput(value=throughput)
-        logging.warning('perf_analyzer output did not contain throughput.')
-
-    def _parse_perf_latency_avg(self, section):
-        """
-        Parses avg latency from the perf output
-        """
-        return self._parse_perf_latency(section, "Avg latency", PerfLatencyAvg)
-
-    def _parse_perf_latency_p90(self, section):
-        """
-        Parses p90 latency from the perf output
-        """
-        return self._parse_perf_latency(section, "p90 latency", PerfLatencyP90)
+        with open(self._config['model-name'], mode='r') as f:
+            csv_reader = csv.DictReader(f, delimiter=',')
 
-    def _parse_perf_latency_p95(self, section):
-        """
-        Parses p95 latency from the perf output
-        """
-        return self._parse_perf_latency(section, "p95 latency", PerfLatencyP95)
-
-    def _parse_perf_latency_p99(self, section):
-        """
-        Parses p99 latency from the perf output
-        """
-        return self._parse_perf_latency(section, "p99 latency", PerfLatencyP99)
-
-    def _parse_perf_latency(self, section, latency_name=None, parser_cls=None):
-        """
-        Parses latency from the perf output
-        """
-        if latency_name == None and parser_cls == None:
-            # Defaults to p99 latency
-            latency_name = "p99 latency"
-            # Different from PerfLatencyP99
-            parser_cls = PerfLatency
-
-        latency = re.search(latency_name + ": (\d+\.\d+|\d+)", section)
-        if latency:
-            latency = float(latency.group(1)) / 1e3
-            return parser_cls(value=latency)
-        logging.warning("perf_analyzer output did not contain " + latency_name +
-                        ".")
-
-    def _parse_perf_server_queue(self, section):
-        """
-        Parses serve queue time from the perf output
-        """
+            for row in csv_reader:
+                self._perf_records = self._extract_metrics_from_csv(
+                    metrics, row)
 
-        server_queue = re.search('queue (\d+) usec', section)
-        if server_queue:
-            server_queue = float(server_queue.group(1)) / 1e3
-            return PerfServerQueue(value=server_queue)
-        logging.warning(
-            'perf_analyzer output did not contain server queue time.')
+        os.remove(self._config['model-name'])
 
-    def _parse_perf_server_compute_input(self, section):
+    def _extract_metrics_from_csv(self, requested_metrics, csv_metrics):
+        """ 
+        Extracts the requested metrics from the CSV and creates a list of Records
```

Comment:
i guess technically a single line of a CSV is a CSV, but this tripped me up at first with the naming since the function implies that it operates on an entire file instead of just one line.

Reply:
Replaced csv with row